sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +380,487 @@ def transform(self, X):
+            `abs_correlation_matrix` can have nans, which will be replaced
+            with `tolerance`.
+
+        Returns
+        -------
+        abs_correlation_matrix : array-like, shape (n_features, n_features)
+            Absolute correlation matrix of X at the beginning of the current
+            round. The diagonal has been zeroed out and each feature has been
+            normalized to sum to 1.
+        """
+        # at each stage all but one of the features is used as input
+        n_features = X_filled.shape[1]
+        if (self.n_nearest_features is None or
+                self.n_nearest_features > n_features - 1):
+            return None
+        abs_correlation_matrix = np.abs(np.corrcoef(X_filled.T))
+        # np.corrcoef is not defined for features with zero std
+        abs_correlation_matrix[np.isnan(abs_correlation_matrix)] = tolerance
+        np.fill_diagonal(abs_correlation_matrix, 0)
+        abs_correlation_matrix = normalize(abs_correlation_matrix,
+                                           norm='l1',
+                                           axis=0)
+        return abs_correlation_matrix
+
+    def fit_transform(self, X, y=None, **fit_params):
+        """Fits the imputer on X and return the transformed X.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Input data, where "n_samples" is the number of samples and
+            "n_features" is the number of features.
+
+        Returns
+        -------
+        X : array-like, shape (n_samples, n_features)
+            The imputed input data.
+        """
+        if self.estimator is None:
+            from ..linear_model import BayesianRidge
+            self.estimator_ = BayesianRidge()
+        else:
+            self.estimator_ = clone(self.estimator)
+
+        # check that the estimator's predict method has return_std argument
+        if 'return_std' not in signature(self.estimator_.predict).parameters:
+            raise ValueError(
+                "The regression estimator is %s and its predict method does "
+                "not support 'return_std=True'. This is required for MICE." %
+                type(self.estimator_)
+            )
+
+        # pre-processing of X
+        X = check_array(X, dtype=np.float32, order="F", force_all_finite=False)
+
+        # parse min and max values
+        self.min_value_ = np.nan if self.min_value is None else self.min_value
+        self.max_value_ = np.nan if self.max_value is None else self.max_value
+
+        # initial imputation
+        mask_missing_values = _get_mask(X, self.missing_values)
+        self.initial_imputer_ = Imputer(missing_values=self.missing_values,
+                                        strategy=self.initial_strategy,
+                                        axis=0)
+        X_filled = self.initial_imputer_.fit_transform(X)
+
+        valid_mask = np.logical_not(np.isnan(
+            self.initial_imputer_.statistics_
```

Comment:
try to improve the readability; jump a line before np.isnan and put everything on the same line.

Reply:
Done.